sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -23,8 +23,12 @@ static Command *getCommand(const EventImplPtr &Event) {
 
 std::vector<EventImplPtr>
 Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
-  std::vector<EventImplPtr> Result;
   Command *Cmd = getCommand(Event);
+  // We can't iterate through the empty command. It can be empty if Event was
+  // created with default constructor and was not initialized after)
```

Comment:
1. Not only using default constructor but interoperability constructor as well. 
2. Not sure "not initialized after" is a correct wording here as I do not see any way to alter already created event in any way.

Reply:
1. not sure what is interoperability constructor 
2. there is a method setCommand for event, so it can be changed

Reply:
#1
The SYCL spec says event class has the following public c'tors:
event(); // <== default
event(cl_event clEvent, const context& syclContext); // <== interoperability one

#2
setCommand is internal method, I think it would be nice to have comment here that explains which user code will cause Command to be nullptr. Because the next question I would ask reading this comment is why command is not initialized later, when this can happen.